### PR TITLE
Allwinner: Add Beelink GS1

### PR DESCRIPTION
**Pull request by clementperon**
Beelink GS1 is an Allwinner H6 Board. Mainlined since Kernel 5.1 and U-Boot 2019.07.

The U-boot bump will conflict with : https://github.com/LibreELEC/LibreELEC.tv/pull/3703

What's working:
Display (1024x768max)
USB2/3
Ethernet

Doesn't work:
Audio HDMI
SPDIF (Merged in 5.3)
InfraRed (Merged in 5.4)
Wi-Fi
Bluetooth
RTC
Thermal

### CODE REVIEW

File: projects/Allwinner/devices/H6/patches/u-boot/002-orange-pi-3-support.patch [hunk 1]
```diff
@@ -1,363 +0,0 @@
-diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
```
Comment (jernejsk) [inline projects/Allwinner/devices/H6/patches/u-boot/002-orange-pi-3-support.patch]: Why did you delete this file? It's not in v2019.7, so I guess it needs just slight update.

File: projects/Allwinner/devices/H6/patches/u-boot/002-orange-pi-3-support.patch [hunk 2]
```diff
@@ -1,363 +0,0 @@
...
-@@ -0,0 +1,13 @@
-+CONFIG_ARM=y
-+CONFIG_ARCH_SUNXI=y
-+CONFIG_SPL=y
-+CONFIG_MACH_SUN50I_H6=y
-+CONFIG_MMC0_CD_PIN="PF6"
-+# CONFIG_PSCI_RESET is not set
-+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
-+CONFIG_NR_DRAM_BANKS=1
-+# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
```
Comment (jernejsk) [inline projects/Allwinner/devices/H6/patches/u-boot/002-orange-pi-3-support.patch]: For v2019.7, all Allwinner U-Boot configs needs `CONFIG_SPL_TEXT_BASE` (0x20060 for H6), please add it when you'll update this patch.

Please add additional U-Boot patch for Beelink GS1 DT and add emac node and ethernet0 alias. It doesn't have to work, but it must be there in order to have a constant MAC address.

File: projects/Allwinner/devices/H6/patches/u-boot/002-orange-pi-3-support.patch (hunk 1 above)
Comment (clementperon) [reply projects/Allwinner/devices/H6/patches/u-boot/002-orange-pi-3-support.patch]: Doesn't apply and forget to fix it before submiting the PR

File: projects/Allwinner/devices/H6/patches/u-boot/002-orange-pi-3-support.patch (hunk 2 above)
Comment (clementperon) [reply projects/Allwinner/devices/H6/patches/u-boot/002-orange-pi-3-support.patch]: OK will do

**Comment by clementperon** [comment]
Ok with DDC regulator sound and display are now OK :)

**Comment by jernejsk** [comment]
This is almost ready to merge. Just two small nitpicks:
1. Please put changes from scripts/uboot_helper to separate commit with title "uboot_helper: Add Beelink GS1 board" or something similar
2. Put fixes for A64 and H6 DRAM to separate PR. They are not GS1 specific.

**Comment by clementperon** [comment]
@jernejsk Done
